ssl/s3_lib.c

```diff
@@ -4723,19 +4723,33 @@ EVP_PKEY *ssl_generate_pkey_group(SSL *s, uint16_t id)
      */
 # ifndef OPENSSL_NO_DH
     if (gtype == TLS_GROUP_FFDHE)
+#  if 0
+        pctx = EVP_PKEY_CTX_new_from_name(s->ctx->libctx, "DH", s->ctx->propq);
+#  else
         pctx = EVP_PKEY_CTX_new_id(EVP_PKEY_DH, NULL);
+#  endif
 #  ifndef OPENSSL_NO_EC
     else
-#  endif
-# endif
+#  endif /* OPENSSL_NO_EC */
```

Comment:
We appear to have an ifndef/else/endif trio with no content in either branch?

Reply:
No - we have ifndef/endif with "else" as the content!!

Reply:
Totally not confusing :)
Thanks for the cluebat.